src/signed-data-store/signed-data-store.ts

```diff
@@ -0,0 +1,130 @@
+  if (Number.parseInt(timestamp, 10) * 1000 > Date.now() + 60 * 60 * 1000) {
+    logger.warn(
+      `Refusing to store sample as timestamp is more than one hour in the future: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  if (Number.parseInt(timestamp, 10) * 1000 > Date.now()) {
+    logger.warn(
+      `Sample is in the future, but by less than an hour, therefore storing anyway: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+  }
+
+  return true;
+};
+
+export const checkSignedDataIntegrity = (signedData: SignedData) => {
+  const { airnode, templateId, timestamp, encodedValue } = signedData;
+
+  if (!checkTimestamp(signedData)) {
+    return;
+  }
+
+  if (verifySignedData(signedData)) {
+    logger.warn(
+      `Refusing to store sample as signature does not match: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  return true;
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const setStoreDataPoint = async (signedData: SignedData) => {
+  const { airnode, templateId, signature, timestamp, encodedValue } = signedData;
+
+  if (!checkSignedDataIntegrity(signedData)) {
+    logger.warn(`Signed data received from signed data API has a signature mismatch.`);
+    logger.warn(JSON.stringify({ airnode, templateId, signature, timestamp, encodedValue }, null, 2));
+    return;
+  }
+
+  if (!signedApiStore[airnode]) {
+    signedApiStore[airnode] = {};
+  }
+
+  const existingValue = signedApiStore[airnode]![templateId];
+  if (existingValue && existingValue.timestamp >= timestamp) {
+    logger.debug('Skipping store update. The existing store value is fresher.');
+    return;
+  }
+
+  logger.debug(
+    `Storing sample for (Airnode ${airnode}) (Template ID ${templateId}) (Timestamp ${new Date(
+      Number.parseInt(timestamp, 10) * 1000
+    ).toISOString()}), ${BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8}`
+  );
+
+  signedApiStore[airnode]![templateId] = { signature, timestamp, encodedValue };
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const getStoreDataPoint = async (airnode: AirnodeAddress, templateId: TemplateId) =>
```

Comment:
I think that the requirements for Airseeker storage are quite small and we can be quite sure to stick with in-memory storage. 

This might not be true for signed API which needs to persist multiple data points for each dAPI (because it can serve delayed data as well) so we opted for "async" in-memory store. But even that is questionable and maybe premature.